Tasks/SonarQubePostTest/SonarQubeBuildBreaker.ps1

```diff
@@ -0,0 +1,176 @@
+. ./SonarQubeHelper.ps1
+
+$DefaultNumRetriesForAnalysisToComplete = 120; # there 1s delay between each retry plus the time it takes to make a web call
+
+#
+# Top-level orchestrating logic
+# 
+function BreakBuildOnQualityGateFailure
+{
+    
+    $breakBuild = GetTaskContextVariable "MsBuild.SonarQube.BreakBuild"            
+    $breakBuildEnabled = Convert-String $breakBuild Boolean    
+
+    if ($breakBuildEnabled)
+    {
+        if (IsPrBuild)
+        {
+            Write-Warning "Ignoring the setting of breaking the build on quality gate failure because the build was triggered by a pull request."
+            return;
+        }
+       
+        $analysisId = WaitForAnalysisToFinish
+        $qualityGateStatus = QueryQualityGateStatus $analysisId
+        FailBuildOnQualityGateStatus $qualityGateStatus
+
+    }
+    else
+    {
+        Write-Host "The build was not set to fail if the associated quality gate fails."
+    }
+}
+
+#
+# Reads the task id from task-report.txt that is dropped by the sonar-scanner to give the task id
+#
+function FetchTaskIdFromReportFile
+{
+    param ([string]$reportTaskFile)
+
+    $content = [System.IO.File]::ReadAllText($reportTaskFile);
+    $matchResult = [System.Text.RegularExpressions.Regex]::Match($content, "ceTaskId=(.+)");    
```

Comment:
Is this a single line match? Can you be more specific e.g. by anchoring the reg ex to the start/end of the line to reduce the change of it matching anything else?